src/carbon/adapters/smtp_adapter.cr

```diff
@@ -2,7 +2,7 @@ class Carbon::SmtpAdapter < Carbon::Adapter
   Habitat.create do
     setting host : String = "localhost"
     setting port : Int32 = 25
-    setting helo_domain : String? = nil
+    setting helo_domain : String
```

Comment:
I am not too happy with this. I see that is it a quick fix for compatibility with the new email shard, but I do not think _this_ shard should require the user to configure a `helo_domain`. I cannot remember ever having to have to set one when configuring an email client (which an application using carbon would be in this case).

Also, a `helo_domain` will not be among the usual items when you ask your email provider / it department for email credentials.

Thus I think we should provide a sensible default here. Maybe just like the email shard did? See https://github.com/arcage/crystal-email/blob/v0.6.4/src/email/client.cr#L93